component/file_cache/file_cache.go

```diff
@@ -691,69 +691,99 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) DownloadFile(options internal.DownloadFileOptions) (*handlemap.Handle, error) {
+
+	flag := 0
+	//extract flag out of the values from handle
+	flags, found := options.Handle.GetValue("flag")
+	if found {
+		flagsStruct, ok := flags.(struct{ flags int })
+		if !ok {
+			log.Err("FileCache::getHandleData : error Type assertion failed on getting flag for %s", options.Handle.Path)
+			return options.Handle, fmt.Errorf("type assertion failed on getting flag forfor %s", options.Handle.Path)
+		}
+		flag = flagsStruct.flags
+	}
```

Comment:
The lack of flag means that it is cached. The flag in place means it needs to be downloaded

Reply:
Yes, but if it does not have a flag, then it seems like you would not need to continue with the rest of the function since you would know here the file is cached. Also, I am not sure how you use the flag variable here. You set flag := 0 and in the case it is not found, then it is set to 0 and later passed into an OpenFile call, which seems incorrect in the case where the file is already cached.

Reply:
I was letting it download anyway due to some tests that were now calling DownloadFile() directly instead of OpenFile() as a way to download. the TestHardLimitOnSize was failing because originally, it was returning a nil handle once it failed to find the values in the handle.

I could have called OpenFile() before calling DownloadFile() in the test to potentially remedy this. Would that be a more preferable approach?

For the flag, what is a better default value to provide? I think I actually need to use the os.O_RD instead of a hard number like I was doing.

Reply:
Yes, I think having the test do OpenFile and then Download is preferable. We shouldn't change out code to make the tests pass. And every time we interact with a file OpenFile is always called first, so that is a reasonable change to this test, and what it was doing before since OpenFile did previously download the file. And I don't think you need a default for the flag here, if you find the flag, then you use that value. If you don't then it didn't need to be downloaded anyway.